dataactvalidator/config/sqlrules/a18_cross_file.sql

```diff
@@ -1,10 +1,26 @@
 SELECT
-    approp.row_number,
-    approp.tas,
-    approp.gross_outlay_amount_by_tas_cpe,
-    SUM(op.gross_outlay_amount_by_pro_cpe) as gross_outlay_amount_by_pro_cpe_sum
-FROM appropriation approp JOIN object_class_program_activity op
-ON approp.tas = op.tas AND approp.submission_id = op.submission_id
+	approp.row_number,
+	approp.allocation_transfer_agency,
+	approp.agency_identifier,
+	approp.beginning_period_of_availa,
+	approp.ending_period_of_availabil,
+	approp.availability_type_code,
+	approp.main_account_code,
+	approp.sub_account_code,
+	approp.gross_outlay_amount_by_tas_cpe,
```

Comment:
The answer to the `select` and `group by` part of your question: https://github.com/fedspendingtransparency/data-act-broker-backend/pull/207#discussion_r74628071

The `group by` fields need to match the fields in the `select`. 

To your last question, `approp.row_number` is a bit misleading--it refers to the row number on the .csv submitted by the user, so it is unique within a submission but not within the entire table.

Good questions, all!